### PR TITLE
Updated Read/Write & Code Standard

### PR DESCRIPTION
**Pull request by Ershany-IBM**
- Updated the JVM Image header to properly read/write the class tables
- Updated all of the code for our feature, to follow the OMR standard

### CODE REVIEW

**Comment by Ershany-IBM** [comment]
OMR Code Standard:
https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md

File: runtime/vm/JVMImage.cpp [hunk 1]
```diff
@@ -132,46 +134,49 @@ JVMImage::allocateImageMemory(UDATA size)
...
-
-	if (_jvmImageData->classLoaderTable == 0
-		|| _jvmImageData->classSegmentTable == 0
-		|| _jvmImageData->classPathEntryTable == 0) {
+	WSRP_SET(_jvmImageHeader->classLoaderTable, subAllocateMemory(sizeof(ImageTableHeader)));
+	WSRP_SET(_jvmImageHeader->classSegmentTable, subAllocateMemory(sizeof(ImageTableHeader)));
+	WSRP_SET(_jvmImageHeader->classPathEntryTable, subAllocateMemory(sizeof(ImageTableHeader)));
+
+	if ((0 == _jvmImageHeader->classLoaderTable)
+			|| (0 == _jvmImageHeader->classSegmentTable)
```
Comment (Ershany-IBM) [inline runtime/vm/JVMImage.cpp]: this is a little bit too tabbed in, I will fix it!

File: runtime/vm/JVMImage.cpp [hunk 2]
```diff
@@ -283,28 +288,29 @@ JVMImage::registerEntryInTable(ImageTableHeader *table, UDATA entry)
...
 	uint64_t fileSize = omrfile_flength(fileDescriptor);
-	if (fileSize != sizeof(JVMImageHeader) + imageHeader.imageSize) {
+	if (sizeof(JVMImageHeader) + _jvmImageHeader->imageSize != fileSize) {
 		return false;
 	}
 
-	// Mmap our file into virtual memory
-	JVMImageHeader *block = (JVMImageHeader*)mmap((void*)imageHeader.heapAddress, sizeof(JVMImageHeader) + imageHeader.imageSize, PROT_READ, MAP_PRIVATE, fileDescriptor, 0);
-	_heap = (J9Heap*)(block + 1);
+	JVMImageHeader *block = (JVMImageHeader *)mmap(
```
Comment (akshayben) [inline runtime/vm/JVMImage.cpp]: `_jvmImageHeader = `

File: runtime/vm/JVMImage.hpp [hunk 3]
```diff
@@ -76,18 +50,51 @@ public
...
+	
+	/*
+	 * Function Members
+	 */
+private:
+protected:
+	void *operator new(size_t size, void *memoryPointer) { return memoryPointer; }
+public:
+	JVMImage(J9JavaVM *vm);
+	~JVMImage(void);
```
Comment (akshayben) [inline runtime/vm/JVMImage.hpp]: My only problem is with the destructor having a void param. I haven't seen it in the codebase anywhere and I couldn't find specifics on coding standards for constructor/destructor

**Comment by akshayben** [comment]
Also for the InitializeJVMImage function I am guessing you would put IMAGE_ERROR = function_call

File: runtime/vm/JVMImage.cpp [hunk 4]
```diff
@@ -283,28 +288,29 @@ JVMImage::registerEntryInTable(ImageTableHeader *table, UDATA entry)
...
 	}
 
-	// Read image header then mmap the rest of the image (heap) into memory
-	JVMImageHeader imageHeader;
-	omrfile_read(fileDescriptor, &imageHeader, sizeof(JVMImageHeader));
+	/* Read image header then mmap the rest of the image (heap) into memory */
+	omrfile_read(fileDescriptor, (void *)_jvmImageHeader, sizeof(JVMImageHeader));
 	uint64_t fileSize = omrfile_flength(fileDescriptor);
-	if (fileSize != sizeof(JVMImageHeader) + imageHeader.imageSize) {
+	if (sizeof(JVMImageHeader) + _jvmImageHeader->imageSize != fileSize) {
```
Comment (akshayben) [inline runtime/vm/JVMImage.cpp]: So for this image size is already defined as sizeof(JVMImageHeader) + heapSize; imageSize =/= heapSize

Comment (Ershany-IBM) [reply runtime/vm/JVMImage.cpp]: Ooooops
Will fix, I will also fix the write function so just makes one "write" function call.

**Comment by Ershany-IBM** [comment]
> 
> 
> Also for the InitializeJVMImage function I am guessing you would put IMAGE_ERROR = function_call

Yes, in this case it is still rvalue == rvalue but it sticks to the previous standard of having the easy to read value on the left!

File: runtime/vm/JVMImage.cpp [hunk 5]
```diff
@@ -283,28 +288,29 @@ JVMImage::registerEntryInTable(ImageTableHeader *table, UDATA entry)
...
+	if (_jvmImageHeader->imageSize != fileSize) {
 		return false;
 	}
 
-	// Mmap our file into virtual memory
-	JVMImageHeader *block = (JVMImageHeader*)mmap((void*)imageHeader.heapAddress, sizeof(JVMImageHeader) + imageHeader.imageSize, PROT_READ, MAP_PRIVATE, fileDescriptor, 0);
-	_heap = (J9Heap*)(block + 1);
+	_jvmImageHeader = (JVMImageHeader *)mmap(
+		(void *)_jvmImageHeader->heapAddress,
+		sizeof(JVMImageHeader) + _jvmImageHeader->imageSize,
```
Comment (akshayben) [inline runtime/vm/JVMImage.cpp]: here too use imageSize instead of `sizeof(JVMImageHeader) + _jvmImageHeader->imageSize`

**Pull request merged**